src/connection/query.ts

```diff
@@ -0,0 +1,75 @@
+import { SQLJob } from "./sqlJob";
+import { QueryResult, ServerResponse } from "./types";
+
+export class Query<T> {
+
+  private static id_ctr: number = 43;
+  private static globalQueryList: Query<any>[] = [];
+  private correlationId: string;
+  private sql: string;
+  private isPrepared: boolean = false;
+  private hasBeenRunAlready: boolean = false;
+  private parameters: any[];
+  private rowsToFetch: number = 100;
+  constructor(private job: SQLJob, private query: string, private parms: any[] = undefined) {
+    this.job = job;
+    this.isPrepared = (undefined !== parms);
+    this.parameters = parms;
+    this.sql = query;
+    Query.globalQueryList.push(this);
+  }
+  public static byId(id: string) {
+    if (undefined === id || '' === id) {
+      return undefined;
+    }
+    for (let srch of Query.globalQueryList) {
+      if (srch.correlationId === id) {
+        return srch;
+      }
+    } return undefined;
+  }
+  public async run(rowsToFetch: number = this.rowsToFetch): Promise<QueryResult<T>> {
+    if (this.hasBeenRunAlready) {

```

Comment:
I think calling `run` twice needs a test case. I personally think this should return an error.